src/ray/gcs/store_client/in_memory_store_client.h

```diff
@@ -73,25 +74,33 @@ class InMemoryStoreClient : public StoreClient {
  private:
   struct InMemoryTable {
     /// Mutex to protect the records_ field and the index_keys_ field.
-    absl::Mutex mutex_;
+    mutable absl::Mutex mutex_;
     // Mapping from key to data.
+    // TODO(dayshah): benchmark reader/writer locks against boost::concurrent_flat_map
     absl::flat_hash_map<std::string, std::string> records_ ABSL_GUARDED_BY(mutex_);
   };
 
-  std::shared_ptr<InMemoryStoreClient::InMemoryTable> GetOrCreateTable(
-      const std::string &table_name) ABSL_LOCKS_EXCLUDED(mutex_);
+  InMemoryTable &GetOrCreateMutableTable(const std::string &table_name);
+
+  // 1) Will return nullptr if the table does not exist.
+  // 2) The returned pointer is valid as long as the InMemoryStoreClient is alive and
+  //    as long as no other thread erases the InMemoryTable from tables_.
```

Comment:
This comment applies to `GetOrCreateMutableTable` as well (can you add to its comment). It's fine for now since we never call `tables_.erase()`.

Reply:
added